lib/logstash/inputs/lumberjack.rb

```diff
@@ -33,7 +33,7 @@ class LogStash::Inputs::Lumberjack < LogStash::Inputs::Base
 
   public
   def register
-    require "lumberjack/server"
+    require "logstash/inputs/lumberjack/server"
```

Comment:
Because the old lumberjack/server was not codec aware, basically with this change we need to create a new codec instance per connection.

This change can be done in the same ```lumberjack/server``` code, but after talking with @jordansissel we agreed that makes more sense to have this logic in here, so I had to redefine the server logic to accept a new codec in this input.